src/Symfony/Component/Config/Resource/DirectoryResource.php

```diff
@@ -87,7 +87,12 @@ public function isFresh($timestamp)
                 continue;
             }
 
-            $newestMTime = max($file->getMTime(), $newestMTime);
+            // early return if any file mtime is greater than passed timestamp
+            if (($fileMTime = $file->getMTime()) > $timestamp) {
+                return false;
+            }
+
+            $newestMTime = max($fileMTime, $newestMTime);
```

Comment:
No, removing that breaks the test `->isFresh() returns false if an existing file is removed`. Yes, though, we could add another condition at the very beginning.